src/main/java/javaslang/collection/List.java

```diff
@@ -133,13 +136,19 @@
             @SuppressWarnings("unchecked")
             final List<T> list = (List<T>) elements;
             return list;
-        } else if (elements instanceof ArrayList || elements instanceof Vector) {
-            @SuppressWarnings("unchecked")
-            final java.util.List<T> indexedList = (java.util.List<T>) elements;
+        } else if (elements instanceof java.util.List<?>) {
```

Comment:
Yes, it would be great if the compiler could check that List<T> is effectively an immutable list. But since it's just an interface, it's impossible to prove that no instance of List<? extends T> could ever violate the statically known type of a List<T>